### PR TITLE
NXP Sabre_iMX6SX_1GB board support added.

### PR DESCRIPTION
**Pull request by mvarecha**

### CODE REVIEW

File: arch/arm/mach-imx/video.c [hunk 1]
```diff
@@ -32,8 +37,14 @@ int board_video_skip(void)
 	}
 
 	if (i < display_count) {
+#if defined(CONFIG_VIDEO_IPUV3)
 		ret = ipuv3_fb_init(&displays[i].mode, displays[i].di ? 1 : 0,
 				    displays[i].pixfmt);
+#elif defined(CONFIG_VIDEO_MXS)
+		ret = mxs_lcd_panel_setup(displays[i].mode,
+					displays[i].pixfmt,
+				    displays[i].bus);
```
Comment (hbeberman) [inline arch/arm/mach-imx/video.c]: U-Boot patch checker: Alignment should match open parenthesis.
```suggestion
					  displays[i].bus);
```

File: arch/arm/mach-imx/video.c [hunk 2]
```diff
@@ -32,8 +37,14 @@ int board_video_skip(void)
 	}
 
 	if (i < display_count) {
+#if defined(CONFIG_VIDEO_IPUV3)
 		ret = ipuv3_fb_init(&displays[i].mode, displays[i].di ? 1 : 0,
 				    displays[i].pixfmt);
+#elif defined(CONFIG_VIDEO_MXS)
+		ret = mxs_lcd_panel_setup(displays[i].mode,
+					displays[i].pixfmt,
```
Comment (hbeberman) [inline arch/arm/mach-imx/video.c]: U-Boot patch checker: Alignment should match open parenthesis.

```suggestion
					  displays[i].pixfmt,
```

File: board/freescale/mx6sxsabresd/mx6sxsabresd.c [hunk 3]
```diff
@@ -139,10 +148,19 @@ static int setup_fec(void)
...
 	imx_iomux_v3_setup_multiple_pads(fec1_pads, ARRAY_SIZE(fec1_pads));
 	setup_fec();
 
-	return cpu_eth_init(bis);
+	ret = cpu_eth_init(bis);
+	if (!(ret)) {
+		ret = fecmxc_initialize_multi(bis, 1, 1, IMX_FEC_BASE);
+		if (ret) {
+			printf("FEC%d MXC: %s:failed\n", 1, __func__);
+		}
```
Comment (hbeberman) [inline board/freescale/mx6sxsabresd/mx6sxsabresd.c]: U-Boot style says to not use braces { } for single statement blocks.

File: board/freescale/mx6sxsabresd/mx6sxsabresd.c [hunk 4]
```diff
@@ -265,9 +291,76 @@ static int setup_lcd(void)
...
+		return;
+	}
+	ret = enable_lvds_bridge(dev->bus);
+	if (ret) {
+		printf("Enable LVDS bridge failed, %d\n", ret);
+		return;
+	}
+
+	imx_iomux_v3_setup_multiple_pads(lvds_ctrl_pads,
+							ARRAY_SIZE(lvds_ctrl_pads));
```
Comment (hbeberman) [inline board/freescale/mx6sxsabresd/mx6sxsabresd.c]: U-Boot patch checker: Alignment should match open parenthesis.
```suggestion
					 ARRAY_SIZE(lvds_ctrl_pads));
```

File: board/freescale/mx6sxsabresd/mx6sxsabresd.c [hunk 5]
```diff
@@ -265,9 +291,76 @@ static int setup_lcd(void)
...
+
+	/* Enable CABC */
+	gpio_request(IMX_GPIO_NR(4, 18), "CABC enable");
+	gpio_direction_output(IMX_GPIO_NR(4, 18) , 1);
```
Comment (hbeberman) [inline board/freescale/mx6sxsabresd/mx6sxsabresd.c]: U-Boot patch checker error: space prohibited before that ','
```suggestion
	gpio_direction_output(IMX_GPIO_NR(4, 18), 1);
```

File: board/freescale/mx6sxsabresd/mx6sxsabresd.c [hunk 6]
```diff
@@ -265,9 +291,76 @@ static int setup_lcd(void)
...
+
+	/* Set Brightness to high */
+	gpio_request(IMX_GPIO_NR(6, 3), "lvds backlight");
+	gpio_direction_output(IMX_GPIO_NR(6, 3) , 1);
```
Comment (hbeberman) [inline board/freescale/mx6sxsabresd/mx6sxsabresd.c]: U-Boot patch checker error: space prohibited before that ','
```suggestion
	gpio_direction_output(IMX_GPIO_NR(6, 3), 1);
```

File: board/freescale/mx6sxsabresd/mx6sxsabresd.c [hunk 7]
```diff
@@ -265,9 +291,76 @@ static int setup_lcd(void)
...
+		.sync           = FB_SYNC_EXT,
+		.vmode          = FB_VMODE_NONINTERLACED
+} }, {
+	.bus = MX6SX_LCDIF1_BASE_ADDR,
+	.addr = 0,
+	.pixfmt = 24,
+	.detect = NULL,
+	.enable	= setup_lcd,
+	.mode	= {
+		.name			= "MCIMX28LCD",
```
Comment (hbeberman) [inline board/freescale/mx6sxsabresd/mx6sxsabresd.c:349]: Alignment with the rest of the list.
```suggestion
		.name           = "MCIMX28LCD",
```

File: board/freescale/mx6sxsabresd/mx6sxsabresd.c [hunk 8]
```diff
@@ -139,10 +148,19 @@ static int setup_fec(void)
...
 int board_eth_init(bd_t *bis)
 {
+	int ret;
+
 	imx_iomux_v3_setup_multiple_pads(fec1_pads, ARRAY_SIZE(fec1_pads));
 	setup_fec();
 
-	return cpu_eth_init(bis);
+	ret = cpu_eth_init(bis);
+	if (!(ret)) {
```
Comment (hbeberman) [inline board/freescale/mx6sxsabresd/mx6sxsabresd.c:157]: Unnecessary parenthesis. Everywhere else just uses `if (!ret)`
```suggestion
	if (!ret) {
```

File: drivers/video/mxsfb.c [hunk 9]
```diff
@@ -34,6 +36,22 @@ __weak void mxsfb_system_setup(void)
 {
 }
 
+static int setup;
+static struct fb_videomode fbmode;
+static int depth;
+
+int mxs_lcd_panel_setup(struct fb_videomode mode, int bpp,
+	uint32_t base_addr)
```
Comment (hbeberman) [inline drivers/video/mxsfb.c]: U-Boot patch checker: Alignment should match open parenthesis.
```suggestion
			uint32_t base_addr)
```

File: include/configs/mx6sxsabresd.h [hunk 10]
```diff
@@ -15,7 +16,7 @@
 #endif
 
 /* Size of malloc() pool */
-#define CONFIG_SYS_MALLOC_LEN		(3 * SZ_1M)
+#define CONFIG_SYS_MALLOC_LEN		(32 * SZ_1M)
```
Comment (hbeberman) [inline include/configs/mx6sxsabresd.h]: This is a pretty big increase over the default. Is it for the framebuffer?

**Review by hbeberman — CHANGES_REQUESTED** [review]
You should run U-Boot's patch checker tool before making pull requests to catch formatting and style violations. `-c #` specifies how many commits you want to check.
```
./tools/patman/patman.py -n -c 1 -t
```

**Review by hbeberman — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**